library/systemd/test/yast2/system_service_test.rb

```diff
@@ -140,43 +238,143 @@
     end
   end
 
-  describe "#start_modes" do
-    before do
-      allow(service).to receive(:socket).and_return(socket)
+  describe "#support_start_on_demand?" do
+    context "when the service has an associated socket" do
+      let(:socket) { service_socket }
+
+      it "returns true" do
+        expect(system_service.support_start_on_demand?).to eq(true)
+      end
     end
 
-    context "when an associated socket exists" do
-      let(:socket) { double("socket", disable: true) }
+    context "when the service has no associated socket" do
+      let(:socket) { nil }
 
-      it "returns :on_boot, :on_demand and :manual" do
-        expect(system_service.start_modes).to eq([:on_boot, :manual, :on_demand])
+      it "returns false" do
+        expect(system_service.support_start_on_demand?).to eq(false)
+      end
+    end
+  end
+
+  describe "#active?" do
+    context "when no action has been performed over the system service" do
+      context "and the underlying service is active" do
+        let(:service_active) { true }
+
+        it "returns true" do
+          expect(system_service.active?).to eq(true)
+        end
+      end
+
+      context "and the underlying service is not active" do
+        let(:service_active) { false }
+        let(:socket) { nil }
+
+        it "returns false" do
+          expect(system_service.active?).to eq(false)
+        end
+      end
+    end
+
+    context "when service was set to be started" do
+      let(:service_active) { false }
+
+      before do
+        system_service.start
+      end
+
+      it "returns true" do
+        expect(system_service.active?).to eq(true)
+      end
+    end
+
+    context "when service was set to be stopped" do
+      let(:service_active) { true }
+
+      before do
+        system_service.stop
+      end
+
+      it "returns false" do
+        expect(system_service.active?).to eq(false)
+      end
+    end
+
+    context "when service was set to be restarted" do
+      let(:service_active) { false }
+
+      before do
+        system_service.restart
+      end
+
+      it "returns true" do
+        expect(system_service.active?).to eq(true)
+      end
+    end
+
+    context "when service was set to be reloaded" do
+      let(:service_active) { false }
+
+      before do
+        system_service.reload
+      end
+
+      it "returns true" do
+        expect(system_service.active?).to eq(true)
       end
     end
+  end
+
+  describe "#keywords" do
+    before do
+      allow(service).to receive(:id).and_return("cups.service")
+    end
 
-    context "when no associated socket exists" do
+    context "when the service has no associated socket" do
       let(:socket) { nil }
 
-      it "returns :on_boot and :manual" do
-        expect(system_service.start_modes).to eq([:on_boot, :manual])
+      it "returns only the service full name" do
```

Comment:
I would specify also type and remove only so "returns array with service full name"